### DISCUSSION TITLE
Grain selection by orientations

### DISCUSSION THREAD

**Discussion by MaelAllard** [mtex-toolbox/mtex#514] [Ask Anything] [answered]
Hi MTex community,

I would like to select groups of grains having a specific crystallographic plane being parallel to the plane of analysis (for example c plane).
I did a beginning of something but got stuck on my way:

`% First search for a grains having the wanted orientation`
`CS=ebsd('Plagioclase').CS`
`gPlag = grains('Plagioclase');`
`gMeanOri=gPlag.meanOrientation.round2Miller; % Get grains mean orientations`
`% condition`
`Indice_001=find(gMeanOri.hkl(:,1) == 0 & gMeanOri.hkl(:,2) == 0 &...`
`gMeanOri.hkl(:,3) == 1)`

Then, as in the documentation I get back to Euler orientations to select my grains:

`% Select one of the grains respecting the condition to get it's Euler orientation`
`ref_temp = gPlag(Indice_001(1))`
`% restrict to the orientation of Miller (001) in Euler space`
`ref_001 = ref_temp.meanOrientation.Euler/degree * Miller(0,0,1,CS,'hkl')`
`% Finally select the corresponding grains allowing 10° variation around (001)`
`grains_oriSelect_001 = gPlag.findByOrientation(gPlag(ref_001).meanOrientation,...`
    `10*degree)`

And here arise the problem, nothing is different between my to variables `grains` and `grains_oriSelect_001`.
I must have done something wrong here but didn't found out where.
Can someone help me please?

Cheers,
Maël

**Comment by ralfHielscher** [comment, marked as answer]
Hi Mael,

please consider asking questions in the forum :) I guess what you looking for is something like

```matlab
c_plane = Miller(0,1,1,cs,'hkl'); % your crystal plane
a_plane = vector3d(0,0,1); % your analysis plane normal

delta = 10*degree; % the threshold
condition = angle(grains.meanOrientation \ a_plane, c_plane) > delta;

grains_selected = grains(ind)
```

All the best,
Ralf.

**Comment by MaelAllard** [comment]
Hi Ralf,

Thanks a lot for your solution,
My apologies for posting this in the wrong platform!

Cheers,
Maël

**Comment by ralfHielscher** [comment]
No problem :)
